File: set1 q2/solution.js
function twoSum(nums, target) {
    const map = new Map();

    for (let i = 0; i < nums.length; i++) {
        const compliment = target - nums[i];
        if (map.has(compliment)) {
            return [map.get(compliment), i];
        }
        map.set(nums[i], i);
    }
    return [];
}


const nums = [2, 7, 11, 15];
const target = 9;
console.log("Indices:", twoSum(nums, target));

// output Indices: [ 0, 1 ] which is correct as 2 + 7 = 9

File: set1 q2/solution.py
def two_sum(nums, target):
    num_map = {}

    for i, num in enumerate(nums):
        complement = target - num

        if complement in num_map:
            return [num_map[complement], i]
        
        num_map[num] = i

    return []

nums = [6, 9, 3, 15]
target = 9
print("Indices:", two_sum(nums, target))

# output is Indices: [0, 2] which is correct as 6 + 3= 9